docs/metrics.md

```diff
@@ -0,0 +1,14 @@
+# Build Controller Metrics
+
+The Build component exposes several metrics to help you monitor the health and behavior of your build resources.
+
+Following build metrics are exposed at service `build-operator-metrics` on port `8383`.
+
+| Name | Type | Description | Status |
+| ---- | ---- | ----------- | ------ |
+| `up_status` | Gauge | The up status of the build-controller. | experimental |
+| `build_count` | Gauge | Build creation count. | experimental |
```

Comment:
This seems to be a counter? If so the type should be counter and the naming best practice is to use the `_total` suffix.

Reply:
Hi @brancz ,

Yes, I considered that before, but in future the build_count maybe dynamically increase or decrease if end user create or delete a build.

That is the plan, so I keep to use Gauge here.

Reply:
Could you explain this in the help text? The convention then would be just `builds`, the `_count` suffix here is going to cause confusion much like I was :)

Reply:
Yes, I agree. Let me change the build and buildrun count to counter type now. If we want to support complex case like increase or decrease, let us change them in future.

Reply:
Sounds good!

Reply:
One addition, if this is about creations, I would recommend to reflect that in the metric name as in: `builds_created_total`

Reply:
Hi @brancz ,

Yes, I followed the official naming `<component_name>_<metric>` in https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/instrumentation.md to change them to:

- build_builds_total
- build_buildruns_total
- build_buildrun_establish_duration_seconds
- build_buildrun_completion_duration_seconds

in the lates commit

Reply:
as this describes the creation, I would recommend to include that in the metric name

Reply:
Yep, @SaschaSchwarze0  also discussed in the below review comment.

Because it is not count from creation, I think it is better:
- build_builds_registered_total
- build_buildruns_completed_total